symforce/slam/imu_preintegration/generate.py

```diff
@@ -0,0 +1,52 @@
+# ----------------------------------------------------------------------------
+# SymForce - Copyright 2022, Skydio, Inc.
+# This source code is under the Apache 2.0 license found in the LICENSE file.
+# ----------------------------------------------------------------------------
+
+from pathlib import Path
+
+from symforce import codegen
+from symforce import typing as T
+from symforce.slam.imu_preintegration.manifold_symbolic import imu_manifold_preintegration_update
+from symforce.slam.imu_preintegration.manifold_symbolic import internal_imu_residual
+
+
+def generate_manifold_imu_preintegration(
+    config: codegen.CodegenConfig, output_dir: T.Openable
+) -> Path:
+    """
+    Generate the on-manifold IMU preintegration update and residual functions.
+    """
+    cg_update = codegen.Codegen.function(
+        imu_manifold_preintegration_update,
+        config=config,
+        output_names=[
+            "new_DR",
+            "new_Dv",
+            "new_Dp",
+            "new_covarinace",
+            "new_DR_D_gyro_bias",
+            "new_Dp_D_gyro_bias",
+            "new_Dp_D_accel_bias",
+            "new_Dv_D_gyro_bias",
+            "new_Dv_D_accel_bias",
+        ],
+    )
+    cg_update.generate_function(output_dir=output_dir, skip_directory_nesting=True)
+
+    cg_residual = codegen.Codegen.function(
+        internal_imu_residual,
+        config=config,
+    ).with_linearization(
+        which_args=[
+            "pose_i",
+            "vel_i",
+            "pose_j",
+            "vel_j",
+            "accel_bias_i",
+            "gyro_bias_i",
+        ]
+    )
+    cg_residual.generate_function(output_dir=output_dir, skip_directory_nesting=True)
+
+    return Path(output_dir)
```

Comment:
Not really sure (perhaps it's conceivable that the output directory might somehow eventually be different?). This was actually how Hayk originally did it. Doing so didn't seem that offensive to me, and I figured if he added it in, then it might not be a bad idea to leave it in.

I'm not particularly attached to it, so I'd be comfortable with removing it.

Reply:
Sounds good. Done.